thoth/user_api/openapi_server.py

```diff
@@ -162,6 +162,13 @@ def _healthiness():
 @app.route("/readiness")
 def api_readiness():
     """Report readiness for OpenShift readiness probe."""
+    try:
+        if not GRAPH.is_schema_up2date():
+            _LOGGER.warning("Database schema is not up to date")
+            return jsonify({"status": "Database schema is not up to date"}), 503, {"ContentType": "application/json"}
+    except DatabaseNotInitialized as exc:
+        _LOGGER.warning("Database is not initialized")
+        return jsonify({"status": str(exc)}), 503, {"ContentType": "application/json"}
```

Comment:
It might be better to keep the service up even if the database is down, uninitialized or the schema is not up to date. Our service can still serve user requests and is much more user friendly to respond with an error in opposite to having the whole system inaccessible.

It's a good practice not to be too strict with liveness probes and readiness probes in the kubernetes world - one can easily put the whole infrastructure down (e.g. [check this article](https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html)).

It might be better idea to remove this part from management api.

Reply:
> It might be better to keep the service up even if the database is down, uninitialized or the schema is not up to date. Our service can still serve user requests and is much more user friendly to respond with an error in opposite to having the whole system inaccessible.
> 
> It's a good practice not to be too strict with liveness probes and readiness probes in the kubernetes world - one can easily put the whole infrastructure down (e.g. [check this article](https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html)).
> 
> It might be better idea to remove this part from management api.

Ah, I see you've already done that. https://github.com/thoth-station/management-api/pull/463

I think this PR can be safely closed. :+1:

Reply:
Thanks for the explanation, I have reopened - https://github.com/thoth-station/management-api/pull/463